optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -334,13 +334,19 @@ The following functions are required for filtering constraint streams of differe
 === Joining
 
 Joining is a way to increase <<constraintStreamsCardinality,stream cardinality>> and it is similar to the inner join
-operation in SQL. As the following figure illustrates, a join creates a cartesian product of the streams being joined:
+operation in SQL. As the following figure illustrates,
+a `join()` creates a cartesian product of the streams being joined:
 
 image::ConstraintStreams/constraintStreamJoinWithoutJoiners.png[align="center"]
 
-Doing this is inefficient because the resulting stream might contain constraint matches that are of no interest to your
-constraint.
-Use `Joiner` to restrict your joins only to the matches you are actually interested in, as shown in this example:
+Doing this is inefficient if the resulting stream contains a lot of constraint matches
+that need to be filtered out immediately.
+
+Instead, use a `Joiner` condition to restrict the joined matches only to those that are interesting:
+
+image::ConstraintStreams/constraintStreamJoinWithJoiners.png[align="center"]
```

Comment:
I switched the order by placing the image before the code, in order to show the concept before code and to have the 2 images vertically closer together for contrast (between with and without a joiner).